### PR TITLE
Go lint fixes

### PR DESCRIPTION
**Pull request by fkneeland-figure**
## Description

This pr: https://github.com/provenance-io/provenance/runs/5380413992?check_suite_focus=true is failing on some lint issues.  We should fix those.

closes: #697 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/metadata/types/events.go [hunk 1]
```diff
@@ -22,26 +22,26 @@ const (
...
 
 	// TLCategory is a string name for labels defining an object category.
-	TLCategory               string            = "category"
-	TLCategory_Entry         TelemetryCategory = "entry"
-	TLCategory_Specification TelemetryCategory = "specification"
-	TLCategory_OSLocator     TelemetryCategory = "object-store-locator"
+	TLCategory              string            = "category"
+	TlcategoryEntry         TelemetryCategory = "entry"
+	TlcategorySpecification TelemetryCategory = "specification"
+	TlcategoryOslocator     TelemetryCategory = "object-store-locator"
```
Comment (dwedul-figure) [inline x/metadata/types/events.go:28]: The `c` in "category" should be capitalized for all these, e.g. `TlCategoryEntry`.

File: x/metadata/types/events.go [hunk 2]
```diff
@@ -22,26 +22,26 @@ const (
...
-	TLType_RecordSpec   TelemetryObjectType = "record-specification"
-	TLType_OSLocator    TelemetryObjectType = "object-store-locator"
+	TLType             string              = "object-type"
+	TltypeScope        TelemetryObjectType = "scope"
+	TltypeSession      TelemetryObjectType = "session"
+	TltypeRecord       TelemetryObjectType = "record"
+	TltypeScopespec    TelemetryObjectType = "scope-specification"
+	TltypeContractspec TelemetryObjectType = "contract-specification"
+	TltypeRecordspec   TelemetryObjectType = "record-specification"
+	TltypeOslocator    TelemetryObjectType = "object-store-locator"
```
Comment (dwedul-figure) [inline x/metadata/types/events.go:38]: The `t` in "type" should be capitalized for all these, e.g. `TlTypeScope`.

File: x/metadata/types/events.go [hunk 3]
```diff
@@ -22,26 +22,26 @@ const (
...
 
 	// TLAction is a string name for labels defining an action taken.
-	TLAction         string          = "action"
-	TLAction_Created TelemetryAction = "created"
-	TLAction_Updated TelemetryAction = "updated"
-	TLAction_Deleted TelemetryAction = "deleted"
+	TLAction        string          = "action"
+	TlactionCreated TelemetryAction = "created"
+	TlactionUpdated TelemetryAction = "updated"
+	TlactionDeleted TelemetryAction = "deleted"
```
Comment (dwedul-figure) [inline x/metadata/types/events.go:44]: The `a` in "action" should be capitalized for all these, e.g. `TlActionCreated`.

**Review by dwedul-figure — COMMENTED** [review]
All of these are basically enum entries; having the underscore makes them easier to read and identify. If you can put a lint ignore statement on them in here and have the linter ignore their uses (in other files) too, that'd really be the best solution.

**Comment by dwedul-figure** [comment]
Looking at the lint action output, it'd be nice if there were a way to have it output the offending file. This is what it looks like to me:
```
Running [/home/runner/golangci-lint-1.[42](https://github.com/provenance-io/provenance/runs/5381959810?check_suite_focus=true#step:4:42).1-linux-amd64/golangci-lint run --out-format=github-actions --timeout 10m] in [] ...
  Error: ReadFile not declared by package os (typecheck)
  Error: WriteFile not declared by package os (typecheck)
  Error: WriteFile not declared by package os (typecheck)
  Error: ReadFile not declared by package os (typecheck)
  Error: "os" imported but not used (typecheck)
  Error: "github.com/cosmos/cosmos-sdk/server" imported but not used (typecheck)
  Warning: var-naming: don't use leading k in Go names; var kErr should be err (revive)
  Warning: var-naming: don't use leading k in Go names; var kErr should be err (revive)
  Warning: var-naming: don't use leading k in Go names; var kErr should be err (revive)
  Warning: var-naming: don't use leading k in Go names; var kErr should be err (revive)
  Warning: var-naming: don't use leading k in Go names; var kErr should be err (revive)
  Warning: var-naming: don't use leading k in Go names; var kErr should be err (revive)
  
  Error: issues found
  Ran golangci-lint in 69963ms
```

Hopefully it's just some sort of `golangci-lint` setting for the config file.

**Comment by dwedul-figure** [comment]
Looks like not having the filenames and line numbers there is a known issue in github actions. It's some sort of interaction between the output and github trying to keep things safe (or something). Here's a link to a comment with a possible workaround: https://github.com/golangci/golangci-lint-action/issues/119#issuecomment-981090648
I'm not sure if `colored-line-number` is the only way to make it work or if `github-actions` will still work.

**Review by arnabmitra — APPROVED** [review]

**Review by dwedul-figure — DISMISSED** [review]
LGTM.

**Review by arnabmitra — APPROVED** [review]

**Review by iramiller — APPROVED** [review]
Nice

**Review by dwedul-figure — APPROVED** [review]
LGTM.

**Pull request merged**